python-lib/gluonts_forecasts/model_handler.py

```diff
@@ -212,41 +111,123 @@ def _convert_parameters_to_class(self, parameters):
         return parameters_converted
 
 
-def list_available_models():
-    """List available models names found in the recipe.json (keys of MODEL_DESCRIPTORS).
+class TrivialIdentity(ModelHandler):
+    model_name = "trivial_identity"
 
-    Returns:
-        dict_keys of model names.
-    """
-    available_models = MODEL_DESCRIPTORS.copy()
-    return available_models.keys()
+    def __init__(self):
+        super().__init__(model_name=self.model_name, label="TrivialIdentity", predictor_class=IdentityPredictor)
+
+    def needs_num_samples(self):
+        return True
+
+    def estimator(self, model_parameters, **kwargs):
+        default_kwargs = {"num_samples": 100}
+        kwargs.update(default_kwargs)
+        return super().estimator(model_parameters, **kwargs)
 
 
-def list_available_models_labels():
-    """List available models labels found in the UI.
+class SeasonalNaive(ModelHandler):
+    model_name = "seasonal_naive"
 
-    Returns:
-        List of model names.
-    """
-    available_models = MODEL_DESCRIPTORS.copy()
-    available_models_labels = []
-    for model in available_models:
-        label = available_models[model].get(LABEL)
-        if label is not None:
-            available_models_labels.append(label)
-    return available_models_labels
-
-
-def get_model_label(model_name):
-    model_descriptor = MODEL_DESCRIPTORS.get(model_name)
-    if model_descriptor is None:
-        return None
-    else:
-        return model_descriptor.get(LABEL, "")
-
-
-def get_model_name_from_label(model_label):
-    available_models = MODEL_DESCRIPTORS.copy()
-    return next(
-        (model_name for model_name in available_models if available_models[model_name].get(LABEL) == model_label), None
-    )
+    def __init__(self):
+        super().__init__(model_name=self.model_name, label="SeasonalNaive", predictor_class=SeasonalNaivePredictor)
+
+    def can_use_seasonality(self):
+        return True
+
+
+class AutoARIMA(ModelHandler):
+    model_name = "autoarima"
```

Comment:
Why not define this directly in the `super().__init__`?

Reply:
We use the model_name as a global property of the ModelHandler because the ModelHandlerRegistry is basically a dict with model_name as keys and ModelHandler() as value and we can register ModelHandler in this dict by doing for example `dict[AutoARIMA.model_name] = AutoARIMA()` instead of `dict["autoarima"] = AutoARIMA()` (so we don't have to repeat the model_name)

Reply:
Ah okay I see, thanks for explaining